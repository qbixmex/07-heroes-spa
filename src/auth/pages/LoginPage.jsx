export const LoginPage = () => {
  return (
    <div className="container py-4">
      <h2>Login Page</h2>
    </div>
  );
};
