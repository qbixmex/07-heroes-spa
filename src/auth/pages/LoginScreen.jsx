import { useContext } from "react";
import { AuthContext } from "../../auth";
import { useNavigate } from "react-router-dom";

export const LoginScreen = () => {

  const navigate = useNavigate();
  const { login } = useContext( AuthContext );

  const handleLogin = () => {
    login('John Doe');
    navigate('/marvel', { replace: true });
  };
  
  return (
    <div className="container py-4">
      <h1>Login Screen</h1>
      <hr />

      <button
        className="btn btn-primary"
        onClick={ handleLogin }
      >
        Login
      </button>
    </div>
  );
};
