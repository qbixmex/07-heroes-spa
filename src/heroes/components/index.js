export * from './HeroesList';
export * from './HeroCard';
export * from './CharactersByHero';
