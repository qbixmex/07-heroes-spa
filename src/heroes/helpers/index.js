export * from './getHeroesByStudio';
export * from './getHeroById';