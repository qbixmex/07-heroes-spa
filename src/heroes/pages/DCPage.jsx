export const DCPage = () => {
  return (
    <h2>DC Page</h2>
  );
};
