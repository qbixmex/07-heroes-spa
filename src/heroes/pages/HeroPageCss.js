export const topCss = 'row animate__animated animate__fadeIn';

export const imageCssClasses = `\
img-fluid \
img-thumbnail \
animate__animated \
animate__fadeInLeft \
mb-4 \
mb-lg-0\
`;
