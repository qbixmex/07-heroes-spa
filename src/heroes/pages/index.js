export * from './DCPage';
export * from './MarvelPage';
export * from './HeroPage';
export * from './SearchPage';
export * from './HeroPageCss';
