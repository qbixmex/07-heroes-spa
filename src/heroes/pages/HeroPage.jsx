import { useMemo } from "react";
import { Navigate, useParams, useNavigate } from "react-router-dom";
import { CharactersByHero } from "../components";
import { getHeroById } from "../helpers";
import { topCss, imageCssClasses } from "./";

export const HeroPage = () => {

  const { id } = useParams();
  const navigate = useNavigate();

  const hero = useMemo(() => getHeroById(id), [ id ]);

  if (!hero) return (<Navigate to='/' />);

  const {
    superhero,
    alter_ego,
    first_appearance,
    characters,
    studio,
  } = hero;

  const onNavigateBack = () => {
    navigate(-1);
  };

  return (
    <>
      <main>
        <div className={ topCss }>
          <div className="col-12 col-lg-5">
            <img
              className={imageCssClasses}
              src={`/assets/heroes/${id}.jpg`}
              alt={superhero}
            />
          </div>
          <div className="col-12 col-lg-7">
            <h1 className="mb-4">{superhero}</h1>
            <h2>Character Information</h2>
            <table className="table">
              <tbody>
                <tr>
                  <th>Alter Ego</th>
                  <td>{alter_ego}</td>
                </tr>
                <tr>
                  <th style={{ width: '160px' }}>First Appearance:</th>
                  <td>{first_appearance}</td>
                </tr>
                <tr>
                  <th>Studio:</th>
                  <td>{studio}</td>
                </tr>
              </tbody>
            </table>
            
            <CharactersByHero characters={characters} alter_ego={alter_ego} />
            <button
              className="btn btn-outline-primary"
              onClick={ onNavigateBack }
            >
              Back
            </button>
          </div>
        </div>
      </main>
    </>
  );
};
