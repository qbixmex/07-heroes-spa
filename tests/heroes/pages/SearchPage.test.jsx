import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { SearchPage } from '../../../src/heroes/pages';

describe('Test on <SearchPage />', () => {
  
  
  test('Should match with Snapshot', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/search']}>
        <SearchPage />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  test('Should show with default values', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/search']}>
        <SearchPage />
      </MemoryRouter>
    );
    const h1 = container.querySelector('h1');
    const searchInput = container.querySelector('#search-text');
    const infoMessage = container.querySelector('#info-message');
    const errorMessage = container.querySelector('#error-message');
    expect(h1.innerHTML).toBe('Search a Hero');
    expect(searchInput).toBeTruthy();
    expect(infoMessage.innerHTML).toContain('Type a hero name to get results');
    expect(infoMessage.className).not.toContain('d-none');
    expect(errorMessage.className).toContain('d-none');
  });

  test('Should show spiderman and input with queryString value', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/search?q=spiderman']}>
        <SearchPage />
      </MemoryRouter>
    );

    const searchInput = container.querySelector('#search-text');
    const errorMessage = container.querySelector('#error-message');
    const card = container.querySelector('.card');
    const cardTitle = container.querySelector('.card-title');
    const image = container.querySelector('[alt=Spiderman]');

    expect(searchInput.value).toBe('spiderman');
    expect(errorMessage.className).toContain('d-none');
    expect(card).toBeTruthy();
    expect(cardTitle.innerHTML).toBe('Spiderman');
    expect(image.src).toContain('/assets/heroes/marvel-spider.jpg');

    // screen.debug();
    
  });
});