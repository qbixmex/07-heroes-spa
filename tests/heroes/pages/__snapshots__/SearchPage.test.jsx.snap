// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <SearchPage /> Should match with Snapshot 1`] = `
<div>
  <h1>
    Search a Hero
  </h1>
  <hr />
  <div
    class="row mt-4"
  >
    <div
      class="col-12 col-lg-5"
    >
      <form>
        <div
          class="row g-2 align-items-center mb-4 mb-lg-0"
        >
          <div
            class="col-12 col-md-10"
          >
            <input
              autocomplete="off"
              class="form-control"
              id="search-text"
              name="searchText"
              placeholder="Search a hero"
              type="text"
              value=""
            />
          </div>
          <div
            class="col-12 col-md-2"
          >
            <div
              class="d-grid"
            >
              <button
                class="btn btn-outline-primary"
                type="submit"
              >
                <i
                  class="bi bi-search"
                />
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
    <div
      class="col-12 col-lg-7"
    >
      <h2
        style="line-height: 28px; margin-bottom: 0px;"
      >
        Results
      </h2>
      <hr />
      <div
        class="alert alert-info animate__animated animate__flash"
        id="info-message"
      >
        <i
          class="bi bi-emoji-wink-fill"
        />
         Type a hero name to get results
      </div>
      <div
        class="alert alert-danger animate__animated animate__flash d-none"
        id="error-message"
      >
        <i
          class="bi bi-emoji-frown-fill"
        />
         There's no hero named 
        <b />
      </div>
    </div>
  </div>
</div>
`;
